beesdoo_base/models/membercard.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+from openerp import models, fields, api
+from random import randint
+import uuid
+
+class MemberCard(models.Model):
+    
+    def _get_current_user(self):
+        return self.env.uid
+    
+    def _get_current_client(self):
+        # TODO : this does not work
+        return self.env['res.partner'].search([('id', '=',self.env.context['active_id'])])
+    
+    def _compute_bar_code(self):
+        rule = self.env['barcode.rule'].search([('name', '=', 'Customer Barcodes')])[0]
+        nomenclature = self.env['barcode.nomenclature']
+        size = 13-len(rule.pattern)
+        ean = rule.pattern + str(uuid.uuid4().fields[-1])[:size] #str(randint(10**(size-1), 10**size-1))
+        code = ean[0:12] + str(nomenclature.ean_checksum(ean))
+        nomenclature.check_encoding(code,'ean13')
```

Comment:
This line return true or false, there is no point to call it, if the result is not check. 
There is no point to check it in real life once we know our code generate valid barcode. 
If we check the result and raise an error the end user cannot do anything anyway.